### PR TITLE
Introduce reverse proxy functionality.

### PR DESCRIPTION
**Pull request by jeremyzahner**
Hi there

I upgraded and tested the new version.

Tested:
- Adding public repositories
- Adding private repositories
- Cron timer & running cron

Also, there now is a functionality for running toran proxy behind a reverse proxy for i.e. SSL termination.

### CODE REVIEW

File: Changelog.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+**1.4.2**
```
Comment (cedvan) [inline Changelog.md]: You update master, replace `**1.4.2**` by `**latest**`

Comment (jeremyzahner) [inline Changelog.md]: Will do.

Comment (jeremyzahner) [inline Changelog.md]: Well actually, what do you mean? Don't you version accordingly to the upstreams versioning?

**Comment by jeremyzahner** [comment]
@cedvan Tagged now.

File: Changelog.md (hunk 1 above)
Comment (cedvan) [inline Changelog.md]: I tag after the merge request directly on master with commit type release :wink:

Comment (cedvan) [inline Changelog.md]: Example : https://github.com/cedvan/docker-toran-proxy/commit/44c4234b64830035114ec44205c66e7a0980e81e

Comment (adamlc) [inline Changelog.md]: @cedvan is this something you can do after merge? These version updates have been really dragging :(

**Comment by cedvan** [comment]
Oh, great job man :smiley:, thank you!

I will update this week-end!

**Comment by adamlc** [comment]
Can this be be merged now @cedvan?

File: assets/vhosts/toran-proxy-http-reverse.conf [hunk 2]
```diff
@@ -0,0 +1,21 @@
+server {
```
Comment (cedvan) [inline assets/vhosts/toran-proxy-http-reverse.conf]: Why not use `toran-proxy-http.conf` ? It the same, no ?

File: scripts/install/toran.sh [hunk 3]
```diff
@@ -51,18 +52,17 @@ ln -s $DATA_DIRECTORY/toran $WORK_DIRECTORY/app/toran
...
-    echo "Installing Token Github..."
+if [ ! -e $DATA_DIRECTORY/toran/composer/auth.json ]; then
     cp -f $ASSETS_DIRECTORY/config/composer.json $DATA_DIRECTORY/toran/composer/auth.json
-    sed -i "s|\"github.com\":|\"github.com\":\"$TORAN_TOKEN_GITHUB\"|g" $DATA_DIRECTORY/toran/composer/auth.json
-else
-    echo "WARNING: "
-    echo "  Variable TORAN_TOKEN_GITHUB is empty !"
-    echo "  You need to setup a GitHub OAuth token because Toran makes a lot of requests and will use up the API calls limit if it is unauthenticated"
-    echo "  Head to https://github.com/settings/tokens/new to create a token. You need to select the public_repo credentials, and the repo one if you are going to use private repositories from GitHub with Toran."
+    if [ "${TORAN_TOKEN_GITHUB}" != "false" ]; then
```
Comment (cedvan) [inline scripts/install/toran.sh:63]: if `TORAN_TOKEN_GITHUB` is set, if file `auth.json` exist, it needs to be updated

File: scripts/install/nginx.sh [hunk 4]
```diff
@@ -5,27 +5,40 @@ echo "Loading Nginx vhosts..."
 rm -f /etc/nginx/sites-enabled/*
 if [ "${TORAN_HTTPS}" == "true" ]; then
 
-    echo "Loading HTTPS Certificates..."
+    if [ "${TORAN_REVERSE}" == "false" ]; then
```
Comment (cedvan) [inline scripts/install/nginx.sh]:
```bash
 if [ "${TORAN_HTTPS}" == "true" ]; then
 
    echo "Loading HTTPS Certificates..."

    ...

    ln -s /etc/nginx/sites-available/toran-proxy-https.conf /etc/nginx/sites-enabled/toran-proxy-https.conf
else
    ln -s /etc/nginx/sites-available/toran-proxy-https-reverse.conf /etc/nginx/sites-enabled/toran-proxy-https-reverse.conf
fi
```

**Comment by cedvan** [comment]
@adamlc, Yes, I need to merge quickly :smiley:

File: scripts/install/toran.sh [hunk 5]
```diff
@@ -51,18 +52,17 @@ ln -s $DATA_DIRECTORY/toran $WORK_DIRECTORY/app/toran
```
Comment (cedvan) [inline scripts/install/toran.sh:58]: Need control `TORAN_REVERSE` to only accept `true` and `false` values

**Comment by cedvan** [comment]
@jeremyzahner, if you can't fix, I merge and I fix after

**Comment by jeremyzahner** [comment]
@cedvan Ill have a look tonight.

**Comment by jeremyzahner** [comment]
I should have done a feature branch for the reverse thing. I'll complete this request tonight.

**Comment by cedvan** [comment]
Ok cool :smiley:

File: assets/vhosts/toran-proxy-http-reverse.conf (hunk 2 above)
Comment (jeremyzahner) [inline assets/vhosts/toran-proxy-http-reverse.conf]: @cedvan Well yes, but i guess there are probably more settings to come that we would need to adjust in a different manner from default http. But i will update that anyway. =)

File: scripts/install/toran.sh (hunk 5 above)
Comment (jeremyzahner) [inline scripts/install/toran.sh:58]: @cedvan Ill do that now.

Comment (jeremyzahner) [inline scripts/install/toran.sh:58]: @cedvan Done now. =)

**Comment by jeremyzahner** [comment]
@cedvan As soon as you merge this i shall create some release branches for you to draft new releases.

**Comment by jeremyzahner** [comment]
@cedvan Ping.

**Comment by cedvan** [comment]
hi @jeremyzahner, 

I had create release 1.3.2. 
Can you fix conflicts ? I merge and create release 1.4.2 after ;)

**Comment by jeremyzahner** [comment]
@cedvan This should work out now. I removed all "versioning" stuff from my commits, so you can create a proper release. Just think about adding the reverse functionality to the changelog when you do a new release. =)

**Comment by cedvan** [comment]
Great !

The procedure for the changelog : Explain the elements added with your PR in the `latest` section, create it if necessary.
Then, when I created a release, I rename the `latest` section with the `release number`

But that's okay I'll do me :wink: !

**Comment by cedvan** [comment]
Thank you for contributing ! :smiley:

**Pull request merged**